### PR TITLE
Astral Armor Fix

### PR DESCRIPTION
**Pull request by Tr1n1ty1**
https://bitbucket.org/looking4group_b2tbc/looking4group/issues/2157/karazhan-curator

Line 129 is optional, both 128 and 129 work fine for me and i'm unable to see a diference, tested this script with 8 ppl about 20 times and worked flawlessly all the time

### CODE REVIEW

File: src/scripts/scripts/zone/karazhan/boss_curator.cpp [hunk 1]
```diff
@@ -72,12 +74,13 @@ struct boss_curatorAI : public ScriptedAI
...
         hatefulBoltTimer = 15000;                           //This time may be wrong
         berserkTimer = 720000;                              //12 minutes
+        astralArmorTimer = 120000;
         enraged = false;
         evocating = false;
         me->ApplySpellImmune(0, IMMUNITY_SCHOOL, SPELL_SCHOOL_MASK_ARCANE, true);
         me->ApplySpellImmune(1, IMMUNITY_STATE, SPELL_AURA_PERIODIC_LEECH, true);
         me->ApplySpellImmune(2, IMMUNITY_STATE, SPELL_AURA_PERIODIC_MANA_LEECH, true);
-
+        m_creature->AddAura(SPELL_ASTRAL_ARMOR, m_creature);
```
Comment (Tr1n1ty1) [inline src/scripts/scripts/zone/karazhan/boss_curator.cpp]: Applied on reset

File: src/scripts/scripts/zone/karazhan/boss_curator.cpp [hunk 2]
```diff
@@ -115,10 +118,19 @@ struct boss_curatorAI : public ScriptedAI
         {
             evocating = true;
             ForceSpellCastWithScriptText(SPELL_EVOCATION, CAST_SELF, SAY_EVOCATE);
+            me->RemoveAurasDueToSpell(SPELL_ASTRAL_ARMOR);
```
Comment (Tr1n1ty1) [inline src/scripts/scripts/zone/karazhan/boss_curator.cpp]: remove during evo

File: src/scripts/scripts/zone/karazhan/boss_curator.cpp [hunk 3]
```diff
@@ -115,10 +118,19 @@ struct boss_curatorAI : public ScriptedAI
...
         if (!enraged && !evocating)
         {
+            if (astralArmorTimer < diff)
+            {
+                ForceSpellCast(SPELL_ASTRAL_ARMOR, CAST_SELF);
+//              m_creature->AddAura(SPELL_ASTRAL_ARMOR, m_creature)  - unsure with this one, both seem to work on my local core
+            }
+            else
+                astralArmorTimer -= diff;
+
```
Comment (Tr1n1ty1) [inline src/scripts/scripts/zone/karazhan/boss_curator.cpp]: reaaply after evo

File: src/scripts/scripts/zone/karazhan/boss_curator.cpp [hunk 4]
```diff
@@ -155,6 +167,8 @@ struct boss_curatorAI : public ScriptedAI
             {
                 enraged = true;
                 ForceSpellCastWithScriptText(SPELL_ENRAGE, CAST_SELF, SAY_ENRAGE);
+                me->RemoveAurasDueToSpell(SPELL_ASTRAL_ARMOR);
+
```
Comment (Tr1n1ty1) [inline src/scripts/scripts/zone/karazhan/boss_curator.cpp]: remove in enrage

**Comment by Looking4Group** [comment]
We will test that. Proof seems to be legit. Thanks.

**Comment by TrenchFox** [comment]
Hey I am going to test this within the hour and report back if it's working g properly.

For future reference please submit a video with the suggested fix it helps prove the validity of this fix.

Thanks for your time,.

**Comment by Tr1n1ty1** [comment]
Source for the fix is in the Bitbucket Link, is he applying the Spell to himself? For me he does and he takes reduced dmg, if he applies but doesn't take reduced maybe the Spell itself is bugged in HG Core

**Comment by TrenchFox** [comment]
Maybe so, I would suggest that you bring up a version of Hellground Core or Looking4Group Core.

I cant tell if he is applying the spell to himself, nothing is showing up as a buff on the boss. I tried AddAuras and ForceSpellCast both of them were not applying any buff, nor were they reducing the amount of damage that the boss took.

You might be onto something suggesting that the spell might be bugged.

Could you send a comment with the "Astral Armor" entry in dbc_spell?

**Comment by Xadras** [comment]
hm i've got the same problem as trenchfox - will have a deeper look later

**Comment by Xadras** [comment]
We found out, that curator is arcane immune and this spell is an arcane spell.
We changed it to be physical and it worked. Thanks for your work!

**Comment by TrenchFox** [comment]
Thanks for the help, your pull request highly influenced my fix. Kudos to Tr1n1t1

**Comment by Tr1n1ty1** [comment]
sure nP, was away till now, but yes the spell shoud've already been physical as it was in my core

**Pull request closed without merging**